File: src/publisher/src/publisher_node.cpp
#include <ros/ros.h>
#include <geometry_msgs/Twist.h> 


int main(int argc, char **argv)
{
 ros::init(argc, argv, "publisher_node");  
 ros::NodeHandle n; 
 ros::Publisher pub = n.advertise<geometry_msgs::Twist>("turtle1/cmd_vel",1000);
 ros::Rate loop_rate(5);
 while(ros::ok()) 
 {
 geometry_msgs::Twist vel; 
 vel.linear.x = 1; 
 vel.angular.z = 1;
 pub.publish(vel);  
 ROS_INFO_STREAM("publishing velocity command: " << "linear (x)=" << vel.linear.x << ", angular (z)=" << vel.angular.z);  
 loop_rate.sleep();
 }
 return 0;
}

File: src/publisher/scripts/publisher_node.py
#!/usr/bin/env python
import rospy
from geometry_msgs.msg import Twist

if __name__ == '__main__':
 pub = rospy.Publisher('turtle1/cmd_vel',Twist,queue_size=1000)
 rospy.init_node('publisher_node')
 loop_rate = rospy.Rate(5)

 while not rospy.is_shutdown():
	 vel=Twist()
	 vel.linear.x = 1.0
	 vel.angular.z = 1.0
	 pub.publish(vel)
	 rospy.loginfo("publishing velocity command: linear (x)=%4.2f, angular (z)=%4.2f",vel.linear.x,vel.angular.z);
	 loop_rate.sleep()
